app/utils/injectReducer.js

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import hoistNonReactStatics from 'hoist-non-react-statics';
+
+import getInjectors from './reducerInjectors';
+
+/**
+ * Dynamically injects a reducer
+ *
+ * @param {string} key A key of the reducer
+ * @param {function} reducer A reducer that will be injected
+ * @param {string} [mode] By default (constants.DAEMON) the reducer is injected on component mount and is never removed.
+ * Another option is constants.RESTART_ON_REMOUNT—the reducer is started when component mounts and is removed on
+ * component un-mount for improved performance.
+ *
+ */
+export default ({ key, reducer, mode }) => (WrappedComponent) => {
```

Comment:
Sometimes we have more than one reducers, so param should be an array of `{ key, reducer, mode }`, eg: `[{key1, reducer1, mode1}, {key2, reducer2, mode2}]`